substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVSectionImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.objectfile.pecoff.cv;
+
+import com.oracle.objectfile.BasicProgbitsSectionImpl;
+import com.oracle.objectfile.BuildDependency;
+import com.oracle.objectfile.LayoutDecision;
+import com.oracle.objectfile.LayoutDecisionMap;
+import com.oracle.objectfile.ObjectFile;
+import com.oracle.objectfile.pecoff.PECoffObjectFile;
+import org.graalvm.compiler.debug.DebugContext;
+
+import java.util.Map;
+import java.util.Set;
+
+abstract class CVSectionImpl extends BasicProgbitsSectionImpl {
+
+    boolean debug = false;
+    long debugTextBase = 0;
+    long debugAddress = 0;
+    int debugBase = 0;
+
+    CVSectionImpl() {
+    }
+
+    @Override
+    public void setElement(ObjectFile.Element e) {
+        super.setElement(e);
+        /* define the section as a COFF symbol */
+        getOwner().createDefinedSymbol(getSectionName(), getElement(), 0, 0, false, false);
+    }
+
+    private String debugSectionLogName() {
+        /*
+         * Use prefix cv4 plus the section name (which already includes a dot separator) for the
+         * context key. For example messages for type section will be keyed using "cv4.debug$T".
+         * Other info formats use their own format-specific prefix.
+         */
+        assert getSectionName().startsWith(".debug$");
```

Comment:
Actually, it's just the prefix, because there are (at least) several section names used by CV4 (I currently implement only two - symbols and types).  I have added a prefix constant in CVConstants and will use it here.

Reply:
Ah yes, sorry. I missed the extra S at the end of CV_SYMBOL_SECTION_NAME and likewise CV_TYPE_SECTION_NAME ending with T. Adding a final static field for the prefix and using it here as well as to define the other symbols is a good idea.